observer/probers/tls/tls_conf.go

```diff
@@ -42,15 +45,28 @@ func (c TLSConf) UnmarshalSettings(settings []byte) (probers.Configurer, error)
 }
 
 func (c TLSConf) validateHostname() error {
-	url, err := url.Parse(c.Hostname)
+	hostname := c.Hostname
+
+	if strings.Contains(c.Hostname, ":") {
```

Comment:
Couldn't most of these `c.Hostname` aside from line 48 be `hostname` since it's defined?

Reply:
It could, but I think semantically it makes less sense. The point of this is that `c.Hostname` is the _default_ value for `hostname`, but if `c.Hostname` has certain properties, then that default gets overridden. I think leaving it as-is makes it feel more like we're selecting one of two possible values for `hostname`, whereas conditioning on `hostname` itself here makes it feel like we're modifying that variable in-place.